src/routes/adm/helpNumber/update.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @memberof module:router~mainRouter~admRouter
+ * @inner
+ * @namespace helpNumber/update
+ */
+const { HelpNumbers, validateHelpNumbers } = require('../../../models/helpNumbers')
+
+/**
+ * Main update function
+ * @name PATCH /adm/helpNumber/:id
+ * @function
+ * @memberof module:router~mainRouter~admRouter~helpNumber/update
+ * @inner
+ * @async
+ * @param {Object} req
+ * @param {Object} res
+ * @returns 400 if invalid requests
+ * @returns 200 if OK
+ * @returns 422 if helpNumber not found or name already used
+ * @returns 500 if Internal Server Error
+ */
+module.exports = async (req, res) => {
+  try {
+    // Verif received data
+    const id = req.params.id
+
+    if (!id) return res.status(400).json({ message: 'Invalid request' })
+
+    const { error } = validateHelpNumbers(req.body)
+    if (error) {
+      return res.status(400).json({ message: 'Invalid request' })
+    }
+
+    const helpNumberToUpdate = await HelpNumbers.findOne({ _id: id })
+    if (!helpNumberToUpdate) return res.status(422).json({ message: 'Class not found' })
+    const tmp = await HelpNumbers.findOne({
+      name: req.body.name,
+      facility: req.user.facility._id
+    })
+    if (tmp && tmp._id !== id) return res.status(422).json({ message: 'This name is already used' })
+
+    helpNumberToUpdate.name = req.body.name
+    helpNumberToUpdate.telephone = req.body.telephone
+    helpNumberToUpdate.email = req.body.email
+    helpNumberToUpdate.helpNumbersCategory = req.body.helpNumbersCategory
+    helpNumberToUpdate.description = req.body.description
```

Comment:
Need to check if value exist in the body before updating. Otherwise might update the values with undefined if not exist.